test_check_ast_errors fails with Python 3.9

#### Description

A single test fails with Python 3.9. It might be a problem with asteval I guess but I'm not sure (I'm just working on getting Python 3.9 supported in Debian, I've never seen these projects before :wink:)

###### A Minimal, Complete, and Verifiable example

```
$ python3.9 -m virtualenv ../venv
$ ../venv/bin/pip install -r requirements-dev.txt 
$ ../venv/bin/python -m pytest
```

###### Error message:

```
============================================================================================================================== test session starts ===============================================================================================================================
platform linux -- Python 3.9.0rc1, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/mwhudson/src/pkg/plus_one/lmfit-py
collected 545 items                                                                                                                                                                                                                                                              

tests/test_1variable.py .                                                                                                                                                                                                                                                  [  0%]
tests/test_NIST_Strd.py ...........................                                                                                                                                                                                                                        [  5%]
tests/test_algebraic_constraint.py ....                                                                                                                                                                                                                                    [  5%]
tests/test_ampgo.py ............                                                                                                                                                                                                                                           [  8%]
tests/test_basicfit.py .                                                                                                                                                                                                                                                   [  8%]
tests/test_basinhopping.py .....                                                                                                                                                                                                                                           [  9%]
tests/test_bounded_jacobian.py .                                                                                                                                                                                                                                           [  9%]
tests/test_bounds.py .                                                                                                                                                                                                                                                     [  9%]
tests/test_brute.py ............                                                                                                                                                                                                                                           [ 11%]
tests/test_builtin_models.py .......                                                                                                                                                                                                                                       [ 13%]
tests/test_confidence.py .............                                                                                                                                                                                                                                     [ 15%]
tests/test_covariance_matrix.py ...........                                                                                                                                                                                                                                [ 17%]
tests/test_custom_independentvar.py .                                                                                                                                                                                                                                      [ 17%]
tests/test_default_kws.py .                                                                                                                                                                                                                                                [ 17%]
tests/test_dual_annealing.py ...                                                                                                                                                                                                                                           [ 18%]
tests/test_itercb.py ..............                                                                                                                                                                                                                                        [ 20%]
tests/test_jsonutils.py ................                                                                                                                                                                                                                                   [ 23%]
tests/test_least_squares.py .....                                                                                                                                                                                                                                          [ 24%]
tests/test_lineshapes.py ...............................................................................                                                                                                                                                                   [ 39%]
tests/test_manypeaks_speed.py .                                                                                                                                                                                                                                            [ 39%]
tests/test_max_nfev.py ...x.................x.x...                                                                                                                                                                                                                         [ 44%]
tests/test_minimizer.py .                                                                                                                                                                                                                                                  [ 44%]
tests/test_model.py ............................................................................s....s..........................................                                                                                                                           [ 67%]
tests/test_model_uncertainties.py ....                                                                                                                                                                                                                                     [ 68%]
tests/test_multidatasets.py .                                                                                                                                                                                                                                              [ 68%]
tests/test_nose.py ................s........                                                                                                                                                                                                                               [ 72%]
tests/test_pandas.py .                                                                                                                                                                                                                                                     [ 73%]
tests/test_parameter.py ..........................................................                                                                                                                                                                                         [ 83%]
tests/test_parameters.py F............................                                                                                                                                                                                                                     [ 88%]
tests/test_printfuncs.py .................................                                                                                                                                                                                                                 [ 95%]
tests/test_saveload.py ..................                                                                                                                                                                                                                                  [ 98%]
tests/test_shgo.py .......                                                                                                                                                                                                                                                 [ 99%]
tests/test_stepmodel.py ..                                                                                                                                                                                                                                                 [100%]

==================================================================================================================================== FAILURES ====================================================================================================================================
_____________________________________________________________________________________________________________________________ test_check_ast_errors ______________________________________________________________________________________________________________________________

    def test_check_ast_errors():
        """Assert that an exception is raised upon AST errors."""
        pars = lmfit.Parameters()
    
        msg = r"at expr='<_ast.Module object at"
        with pytest.raises(NameError, match=msg):
>           pars.add('par1', expr='2.0*par2')

tests/test_parameters.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[NameError("name 'par2' is not defined") raised in repr()] Parameters object at 0x7ff8621f89c0>, name = 'par1', value = None, vary = True, min = -inf, max = inf, expr = '2.0*par2', brute_step = None

    def add(self, name, value=None, vary=True, min=-inf, max=inf, expr=None,
            brute_step=None):
        """Add a Parameter.
    
        Parameters
        ----------
        name : str
            Name of parameter. Must match ``[a-z_][a-z0-9_]*`` and cannot
            be a Python reserved word.
        value : float, optional
            Numerical Parameter value, typically the *initial value*.
        vary : bool, optional
            Whether the Parameter is varied during a fit (default is True).
        min : float, optional
            Lower bound for value (default is ``-numpy.inf``, no lower
            bound).
        max : float, optional
            Upper bound for value (default is ``numpy.inf``, no upper
            bound).
        expr : str, optional
            Mathematical expression used to constrain the value during the
            fit (default is None).
        brute_step : float, optional
            Step size for grid points in the `brute` method (default is
            None).
    
        Examples
        --------
        >>> params = Parameters()
        >>> params.add('xvar', value=0.50, min=0, max=1)
        >>> params.add('yvar', expr='1.0 - xvar')
    
        which is equivalent to:
    
        >>> params = Parameters()
        >>> params['xvar'] = Parameter(name='xvar', value=0.50, min=0, max=1)
        >>> params['yvar'] = Parameter(name='yvar', expr='1.0 - xvar')
    
        """
        if isinstance(name, Parameter):
            self.__setitem__(name.name, name)
        else:
>           self.__setitem__(name, Parameter(value=value, name=name, vary=vary,
                                             min=min, max=max, expr=expr,
                                             brute_step=brute_step))

lmfit/parameter.py:372: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[NameError("name 'par2' is not defined") raised in repr()] Parameters object at 0x7ff8621f89c0>, key = 'par1', par = <[NameError("name 'par2' is not defined") raised in repr()] Parameter object at 0x7ff862149e20>

    def __setitem__(self, key, par):
        """Set items of Parameters object."""
        if key not in self:
            if not valid_symbol_name(key):
                raise KeyError("'%s' is not a valid Parameters name" % key)
        if par is not None and not isinstance(par, Parameter):
            raise ValueError("'%s' is not a Parameter" % par)
        OrderedDict.__setitem__(self, key, par)
        par.name = key
        par._expr_eval = self._asteval
>       self._asteval.symtable[key] = par.value

lmfit/parameter.py:142: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[NameError("name 'par2' is not defined") raised in repr()] Parameter object at 0x7ff862149e20>

    @property
    def value(self):
        """Return the numerical Parameter value, with bounds applied."""
>       return self._getval()

lmfit/parameter.py:819: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[NameError("name 'par2' is not defined") raised in repr()] Parameter object at 0x7ff862149e20>

    def _getval(self):
        """Get value, with bounds applied."""
        # Note assignment to self._val has been changed to self.value
        # The self.value property setter makes sure that the
        # _expr_eval.symtable is kept up-to-date.
        # If you just assign to self._val then _expr_eval.symtable[self.name]
        # becomes stale if parameter.expr is not None.
        if self._expr is not None:
            if self._expr_ast is None:
                self.__set_expression(self._expr)
            if self._expr_eval is not None:
                if not self._delay_asteval:
                    self.value = self._expr_eval(self._expr_ast)
>                   check_ast_errors(self._expr_eval)

lmfit/parameter.py:813: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

expr_eval = <asteval.asteval.Interpreter object at 0x7ff862149be0>

    def check_ast_errors(expr_eval):
        """Check for errors derived from asteval."""
        if len(expr_eval.error) > 0:
>           expr_eval.raise_exception(None)

lmfit/parameter.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asteval.asteval.Interpreter object at 0x7ff862149be0>, node = None, exc = <class 'NameError'>, msg = '', expr = <ast.Module object at 0x7ff862149520>, lineno = None

    def raise_exception(self, node, exc=None, msg='', expr=None,
                        lineno=None):
        """Add an exception."""
        if self.error is None:
            self.error = []
        if expr is None:
            expr = self.expr
        if len(self.error) > 0 and not isinstance(node, ast.Module):
            msg = '%s' % msg
        err = ExceptionHolder(node, exc=exc, msg=msg, expr=expr, lineno=lineno)
        self._interrupt = ast.Raise()
        self.error.append(err)
        if self.error_msg is None:
            self.error_msg = "at expr='%s'" % (self.expr)
        elif len(msg) > 0:
            self.error_msg = msg
        if exc is None:
            try:
                exc = self.error[0].exc
            except:
                exc = RuntimeError
>       raise exc(self.error_msg)
E       NameError: at expr='<ast.Module object at 0x7ff862149520>'

../venv/lib/python3.9/site-packages/asteval/asteval.py:252: NameError

During handling of the above exception, another exception occurred:

    def test_check_ast_errors():
        """Assert that an exception is raised upon AST errors."""
        pars = lmfit.Parameters()
    
        msg = r"at expr='<_ast.Module object at"
        with pytest.raises(NameError, match=msg):
>           pars.add('par1', expr='2.0*par2')
E           AssertionError: Regex pattern "at expr='<_ast.Module object at" does not match "at expr='<ast.Module object at 0x7ff862149520>'".

tests/test_parameters.py:45: AssertionError
------------------------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------------------------
NameError
   <ast.Module object at 0x7ff862149520>
        ^^^
name 'par2' is not defined
---------------------------------------------------------------------------------------------------------------------------- Captured stderr teardown ----------------------------------------------------------------------------------------------------------------------------
NameError
   <ast.Module object at 0x7ff862149520>
        ^^^
name 'par2' is not defined
NameError
   <ast.Module object at 0x7ff862149520>
        ^^^
name 'par2' is not defined
NameError
   <ast.Module object at 0x7ff862149520>
        ^^^
name 'par2' is not defined
NameError
   <ast.Module object at 0x7ff862149520>
        ^^^
name 'par2' is not defined
NameError
   <ast.Module object at 0x7ff862149520>
        ^^^
name 'par2' is not defined
================================================================================================================================ warnings summary ================================================================================================================================
../venv/lib/python3.9/site-packages/past/builtins/misc.py:45
  /home/mwhudson/src/pkg/plus_one/venv/lib/python3.9/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import reload

tests/test_lineshapes.py::test_no_ZeroDivisionError_and_finite_output[donaich]
tests/test_lineshapes.py::test_x_float_value[donaich]
  /home/mwhudson/src/pkg/plus_one/lmfit-py/lmfit/lineshapes.py:299: FutureWarning: Please correct the name of your lineshape function: donaich --> doniach. The incorrect spelling will be removed in a later release.
    warnings.warn(FutureWarning(msg))

tests/test_lineshapes.py::test_x_float_value[step]
tests/test_lineshapes.py::test_x_float_value[rectangle]
  <__array_function__ internals>:5: DeprecationWarning: Calling nonzero on 0d arrays is deprecated, as it behaves surprisingly. Use `atleast_1d(cond).nonzero()` if the old behavior was intended. If the context of this warning is of the form `arr[nonzero(cond)]`, just use `arr[cond]`.

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================================================================================ short test summary info =============================================================================================================================
FAILED tests/test_parameters.py::test_check_ast_errors - AssertionError: Regex pattern "at expr='<_ast.Module object at" does not match "at expr='<ast.Module object at 0x7ff862149520>'".
================================================================================================== 1 failed, 538 passed, 3 skipped, 3 xfailed, 5 warnings in 104.30s (0:01:44) ===================================================================================================
```

###### Version information

```
Python: 3.9.0rc1 (default, Sep  3 2020, 19:31:07) 
[GCC 9.3.0]

lmfit: 1.0.1+74.gbc2975e, scipy: 1.5.3, numpy: 1.19.2, asteval: 0.9.19, uncertainties: 3.1.4
```

###### Link(s)
<!--If you started a discussion on the lmfit mailing list or Stack Overflow, please provide
        the relevant link(s).
-->

--- a/tests/test_parameters.py
+++ b/tests/test_parameters.py
@@ -40,7 +40,7 @@ def test_check_ast_errors():
     """Assert that an exception is raised upon AST errors."""
     pars = lmfit.Parameters()
 
-    msg = r"at expr='<_ast.Module object at"
+    msg = r"at expr='<_?ast.Module object at"
     with pytest.raises(NameError, match=msg):
         pars.add('par1', expr='2.0*par2')
 

